src/cc65/expr.c

```diff
@@ -1015,25 +1015,21 @@ static void ArrayRef (ExprDesc* Expr)
             ** already in Expr. If the base address was a constant, we can even
             ** remove the code that loaded the address into the primary.
             */
-            if (IsTypeArray (Expr->Type)) {
-
-                /* Adjust the offset */
-                Expr->IVal += Subscript.IVal;
-
-            } else {
-
+            if (!IsTypeArray (Expr->Type)) {
+ 
                 /* It's a pointer, so we do have to load it into the primary
                 ** first (if it's not already there).
                 */
-                if (ConstBaseAddr || ED_IsLVal (Expr)) {
+                if (!ConstBaseAddr && ED_IsLVal (Expr)) {
                     LoadExpr (CF_NONE, Expr);
-                    ED_MakeRValExpr (Expr);
+                    ED_FinalizeRValLoad (Expr);
                 }
-
-                /* Use the offset */
-                Expr->IVal = Subscript.IVal;
             }
 
+            /* Adjust the offset */
+            Expr->IVal += Subscript.IVal;
+
+
```

Comment:
Two blank lines?

Reply:
One line removed in d869e3a3b7050be0393dae7d93bcbd5652397c53.